src/strongbox/core.clj

```diff
@@ -761,13 +765,24 @@
        (merge catalogue {:addon-summary-list new-summary-list
                          :total (count new-summary-list)})))))
 
+(defn-spec set-user-catalogue! nil?
+  [new-user-catalogue (s/nilable :catalogue/catalogue)]
+  (let [user-catalogue-idx (some->> new-user-catalogue
+                                    :addon-summary-list
+                                    (map #(select-keys % [:source :source-id]))
```

Comment:
I've repeated myself enough. Push this into utils and hunt down all the other instances of it.